test/src/main/scala/scala/test/twitter_scrooge/BUILD

```diff
@@ -0,0 +1,100 @@
+load("//scala:scala.bzl",   "scala_binary", "scala_library")
+load("//twitter_scrooge:twitter_scrooge.bzl", "scrooge_scala_library")
+
+scrooge_scala_library(
+  name = "thrift1",
+  thrift_library = "//test/src/main/scala/scala/test/twitter_scrooge/thrift",
+  visibility = ["//visibility:public"],
+)
+
+scrooge_scala_library(
```

Comment:
we could make it a build error to not export `scroogeB` in `scala_scrooge_library`. So it would work like this. scrooge_scala_library would have `srcs` that could be thrift files, `deps` that can be thrift_libraries or other `scala_scrooge_library`s and `exports` which can only be other `scala_scrooge_libraries`. We generate code for all the `srcs` and direct `deps`, but we must cover all the transitive `thrift_library` deps with the direct deps of the `exports`, otherwise we have an error (some files would not be owned by a target).

Note, the above includes the ability to add thrifts directly to a `scala_scrooge_library` which would create an ad-hoc thrift-library target and add it to the deps (this makes it nicer to use in the common case of just adding a bunch of thrifts and getting something we can depend on with a scala_library).

What do you think of this UX for BUILD writers?